dotCMS/src/integration-test/java/com/dotcms/filters/interceptor/meta/MetaWebInterceptorTest.java

```diff
@@ -0,0 +1,101 @@
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.StringUtils;
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Unit test for {@link ResponseMetaDataWebInterceptor}
+ * @author jsanca
+ */
+public class MetaWebInterceptorTest {
+
+    private final HttpServletRequest request = mock(HttpServletRequest.class);
+    private final HttpServletResponse response = mock(HttpServletResponse.class);
+
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: {@link ResponseMetaDataWebInterceptor#intercept(HttpServletRequest, HttpServletResponse)}
+     * Given Scenario: calling it adds the header x-dot-server
+     * ExpectedResult: the header is added
+     * @throws IOException
+     */
+    @Test
+    public void check_header_x_dot_server()  {
+
+        final ResponseMetaDataWebInterceptor metaWebInterceptor = new ResponseMetaDataWebInterceptor();
+        final MockHeaderResponse mockHeaderResponse = new MockHeaderResponse(response);
+        metaWebInterceptor.intercept(request, mockHeaderResponse);
+
+        final String header = mockHeaderResponse.getHeader(ResponseMetaDataWebInterceptor.X_DOT_SERVER_HEADER);
+        Assert.assertNotNull(header);
+
+        final Object nodeName =  Try.of(() -> ClusterUtilProxy.getNodeInfo()).
+                getOrElse(Collections.emptyMap()).getOrDefault("friendlyName", "unknown");
+        String tokenHeader = nodeName + StringPool.PIPE + StringUtils.shortify(APILocator.getServerAPI().readServerId(), 10);
+
+        Assert.assertEquals(tokenHeader, header);
+    }
+
+
+    /**
+     * Method to test: {@link ResponseMetaDataWebInterceptor#intercept(HttpServletRequest, HttpServletResponse)}
+     * Given Scenario: calling it does not adds the header x-dot-server
+     * ExpectedResult: the header is not added
+     * @throws IOException
+     */
+    @Test
+    public void check_header_x_dot_server_config_disable()  {
+
+        Config.setProperty(ResponseMetaDataWebInterceptor.RESPONSE_HEADER_ADD_NODE_ID, false);
+        final ResponseMetaDataWebInterceptor metaWebInterceptor = new ResponseMetaDataWebInterceptor();
+        final MockHeaderResponse mockHeaderResponse = new MockHeaderResponse(response);
+        metaWebInterceptor.intercept(request, mockHeaderResponse);
+
+        final String header = mockHeaderResponse.getHeader(ResponseMetaDataWebInterceptor.X_DOT_SERVER_HEADER);
+        Assert.assertNull(header);
+        Config.setProperty(ResponseMetaDataWebInterceptor.RESPONSE_HEADER_ADD_NODE_ID, true);
```

Comment:
if something fails before this line, the property would remain as `false`. Shouldn't this line be in a finally statement?